### PR TITLE
pull_embedding_vector RPC implementation

### LINKED ISSUE #1366: [PS-1] RPC service `pull_embedding_vector` implementation

### PR DESCRIPTION
**Pull request by skydoorkai**
fix #1366

### CODE REVIEW

File: elasticdl/python/tests/pserver_servicer_test.py [hunk 1]
```diff
@@ -175,3 +181,52 @@ def testPullVariable(self):
...
+        req = elasticdl_pb2.Model()
+        req.version = 1
+        req.embedding_table_info.append(embedding_info)
+        embedding_info.name = "layer_b"
+        embedding_info.dim = 16
+        req.embedding_table_info.append(embedding_info)
+        res = self._stub.push_model(req)
+        self.assertEqual(res, empty_pb2.Empty())
+
+        vectors_a_0 = self.get_embedding_vectors("layer_a", id_list_0)
```
Comment (terrytangyuan) [inline elasticdl/python/tests/pserver_servicer_test.py:207]: Reuse the `embedding_info` created earlier?

File: elasticdl/python/tests/pserver_servicer_test.py [hunk 2]
```diff
@@ -175,3 +181,52 @@ def testPullVariable(self):
...
+        embedding_info.dim = 32
+        embedding_info.initializer = "normal"
+
+        id_list_0 = [1, 3, 9, 6]
+        id_list_1 = [8, 9, 1, 0, 6]
+
+        req = elasticdl_pb2.Model()
+        req.version = 1
+        req.embedding_table_info.append(embedding_info)
+        embedding_info.name = "layer_b"
```
Comment (terrytangyuan) [inline elasticdl/python/tests/pserver_servicer_test.py]: Maybe create new variables for this embedding_info? Or add some comments so the logic here is clearer than keep modifying and appending

File: elasticdl/python/ps/servicer.py [hunk 3]
```diff
@@ -52,8 +53,15 @@ def pull_variable(self, request, _):
         return res
 
     def pull_embedding_vector(self, request, _):
-        # TODO: implement this RPC service
-        return elasticdl_pb2.Tensor()
+        ret = elasticdl_pb2.Tensor()
+        if len(request.ids) == 0:
```
Comment (mhaoli) [inline elasticdl/python/ps/servicer.py]: How about `if not request.ids:`?

File: elasticdl/python/tests/pserver_servicer_test.py [hunk 4]
```diff
@@ -82,15 +82,21 @@ def create_default_server_and_stub(self):
             grads_to_wait, lr_staleness_modulation, use_async
         )
 
+    def get_embedding_vectors(self, name, ids):
```
Comment (mhaoli) [inline elasticdl/python/tests/pserver_servicer_test.py:90]: How about renaming this function and telling readers that it is a mock gRPC request? e.g. `mock_request_pull_embedding_vectors`

Comment (skydoorkai) [reply elasticdl/python/tests/pserver_servicer_test.py:90]: This is a true gRPC call.

File: elasticdl/python/tests/pserver_servicer_test.py (hunk 1 above)
Comment (skydoorkai) [reply elasticdl/python/tests/pserver_servicer_test.py:207]: Moved embedding_info definition into `setUp`

File: elasticdl/python/tests/pserver_servicer_test.py (hunk 2 above)
Comment (skydoorkai) [reply elasticdl/python/tests/pserver_servicer_test.py]: Created another embedding info.

**Review by mhaoli — APPROVED** [review]

**Pull request merged**